altimeter_ms5607_spi/altimeter_ms5607_spi.h

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+#if defined(STM32F1)
+#include "stm32f1xx_hal.h"
+#elif defined(STM32F4xx)
+#include "stm32f4xx_hal.h"
+#endif
+
+#ifndef SERIAL_TIMEOUT
+#define SERIAL_TIMEOUT 10
+#endif
+
+class AltimeterMs5607Spi {
+  public:
+    struct Data {
+        // pressure in Pascals
+        double pressure;
+        // temperature in degrees Celsius
+        double temperature;
+        // altitude in meters MSL
+        double altitude;
+    };
+
+    enum State {
+        IDLE,
+        POLL_D1,
+        POLL_D2,
+        COMPLETE,
+        ERROR
+    };
+
+    // oversampling rate setting, see datasheet for more details
```

Comment:
RTFM moment. Can u link the datasheet in a comment somewhere for convenience?

Reply:
the datasheet is in a cdn with a really long nasty url i dont want to link it lol